src/plcc/load_spec/structs.py

```diff
@@ -126,7 +126,9 @@ class RepeatingSyntacticRule(SyntacticRule):
 
 @dataclass
 class SyntacticSpec(list):
-    def __init__(self, rules=None):
+    nonTerminals: list[str]
+    def __init__(self, rules=None, nonTerminals=[]):
```

Comment:
Why are nonTerminals passed as a constructor? Why are they not derived from rules?